tests/models/mistral/test_modeling_mistral.py

```diff
@@ -559,10 +555,6 @@ def test_model_7b_generation(self):
         text = tokenizer.decode(generated_ids[0], skip_special_tokens=True)
         self.assertEqual(EXPECTED_TEXT_COMPLETION[self.cuda_compute_capability_major_version], text)
 
-        del model
-        backend_empty_cache(torch_device)
-        gc.collect()
-
```

Comment:
not help and worse cause some GPU OOM in subsequent tests

Reply:
Happy to have this deleted but very confused why this would cause OOM 😭

Reply:
Got to say I am confusing. `torch.cuda.empty_cache` is not really a magic

> [empty_cache()](https://pytorch.org/docs/stable/generated/torch.cuda.empty_cache.html#torch.cuda.empty_cache) doesn’t increase the amount of GPU memory available for PyTorch.

but I was not expecting it would have undesired side-effect like this (even if it is not helpful).

I don't check if `del model` and `gc.collect()` plays a role here though.

Reply:
Out of curiosity and keep info here for the record:

- it is `test_model_7b_long_prompt` gets OOM.
  - previously with those empty cache, at the beginning of `test_model_7b_long_prompt`, `nvidia-smi` shows `150MiB /  15360MiB` which looks nice but we get OOM afterward  inside this test
  - wihout empty cache, `nvidia-smi` shows `9066MiB/  15360MiB` which looks not great but we **DON'T** get OOM afterward  inside this test

It's very mysterious to me.